module_2/gifts/controller/src/main/java/com/epam/esm/controller/CertificateController.java

```diff
@@ -0,0 +1,111 @@
+package com.epam.esm.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.service.CertificateService;
+
+/**
+ * Controller for working with certificates.
+ * 
+ */
+@RestController
+@RequestMapping("/v1/certificates")
+public class CertificateController {
+	@Autowired
+	private CertificateService certificateService;
```

Comment:
field injection is bad practice. Spring team strongly suggest to use a constructor injection